mllib/src/main/scala/org/apache/spark/ml/attribute/attributes.scala

```diff
@@ -535,3 +541,32 @@ object BinaryAttribute extends AttributeFactory {
     new BinaryAttribute(name, index, values)
   }
 }
+
+/**
+ * An unresolved attribute.
+ */
+private[ml] object UnresolvedAttribute extends Attribute {
```

Comment:
Should this be public, like the other attribute types?  (Not sure if you discussed with @mengxr)